src/I18n/Time.php

```diff
@@ -799,6 +838,100 @@ public static function parseTime($time, $format = null)
     }
 
     /**
+     * Returns the calendar type
+     * @param string|null $locale The locale name in which the date should be displayed (e.g. pt-BR)
+     * @return integer \IntlDateFormatter::TRADITIONAL|\IntlDateFormatter::GREGORIAN
+     */
+    public static function detectCalendarType($locale = null)
+    {
+        if (preg_match('/@calendar=(japanese|buddhist|chinese|persian|indian|islamic|hebrew|coptic|ethiopic)/', $locale)) {
```

Comment:
you dont need the preg_match if null is passed, maybe adding

    if ($locale && ...)

?